src/mappings/points/handlePoints.ts

```diff
@@ -0,0 +1,251 @@
+
+// global vars to count points
+export const pointsCount = new Map<address, Big>()
+
+// keep track of user's LPs
+const perSwapPoolLPCount = new Map<address, Map<SwapPoolId, bigint>>()
+const perBackstopPoolLPCount = new Map<address, Map<BackstopPoolId, bigint>>()
+
+export function addSwapLP(
+    address: address,
+    swapPoolId: SwapPoolId,
+    amount: bigint
+) {
+    if (!perSwapPoolLPCount.has(address)) {
+        perSwapPoolLPCount.set(address, new Map<SwapPoolId, bigint>())
+    }
+    const swapPoolUserLPs = perSwapPoolLPCount.get(address)!
+    if (!swapPoolUserLPs.has(swapPoolId)) {
+        swapPoolUserLPs.set(swapPoolId, BigInt(0))
+    }
+    swapPoolUserLPs.set(swapPoolId, swapPoolUserLPs.get(swapPoolId)! + amount)
+}
+
+export function addBackstopLP(
+    address: address,
+    backstopPoolId: BackstopPoolId,
+    amount: bigint
+) {
+    if (!perBackstopPoolLPCount.has(address)) {
+        perBackstopPoolLPCount.set(address, new Map<BackstopPoolId, bigint>())
+    }
+    const backstopUserLPs = perBackstopPoolLPCount.get(address)!
+    if (!backstopUserLPs.has(backstopPoolId)) {
+        backstopUserLPs.set(backstopPoolId, BigInt(0))
+    }
+    backstopUserLPs.set(
+        backstopPoolId,
+        backstopUserLPs.get(backstopPoolId)! + amount
+    )
+}
+
+export function removeSwapLP(
+    address: address,
+    swapPoolId: SwapPoolId,
+    amount: bigint
+) {
+    if (!perSwapPoolLPCount.has(address)) {
+        return
+    }
+    const swapPoolUserLPs = perSwapPoolLPCount.get(address)!
+    if (!swapPoolUserLPs.has(swapPoolId)) {
+        return
+    }
+    swapPoolUserLPs.set(swapPoolId, swapPoolUserLPs.get(swapPoolId)! - amount)
+}
+
+export function removeBackstopLP(
+    address: address,
+    backstopPoolId: BackstopPoolId,
+    amount: bigint
+) {
+    if (!perBackstopPoolLPCount.has(address)) {
+        return
+    }
+    const backstopUserLPs = perBackstopPoolLPCount.get(address)!
+    if (!backstopUserLPs.has(backstopPoolId)) {
+        return
+    }
+    backstopUserLPs.set(
+        backstopPoolId,
+        backstopUserLPs.get(backstopPoolId)! - amount
+    )
+}
+
+// each block we accumulate the points based on price and LPs that the address has.
+export async function handlePointAccumulation(ctx: Ctx, block: BlockHeader_) {
+    // Cache maps for swap pool prices and backstop pool prices
+    // avoid calling the store on each iteration
+    const swapPoolLPPrices = new Map<SwapPoolId, Big>()
+    const backstopPoolLPPrices = new Map<BackstopPoolId, Big>()
+
+    // also cache swap pools and backstop pools entities to avoid calling the store on each iteration
+    const swapPools = new Map<string, SwapPool>()
+    const backstopPools = new Map<string, BackstopPool>()
+
+    const addresses = new Set<address>([
+        ...perSwapPoolLPCount.keys(),
+        ...perBackstopPoolLPCount.keys(),
+    ])
+
+    // Accumulate points from both swap LPs and backstop LPs
+    for (const address of addresses) {
```

Comment:
Iterate through addresses to more easily return total points this block for each user.